projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -104,6 +104,54 @@ public function add_my_home_menu() {
 		$this->update_menu( 'index.php', 'https://wordpress.com/home/' . $this->domain, __( 'My Home', 'jetpack' ), 'manage_options', 'dashicons-admin-home' );
 	}
 
+	/**
+	 * Adds upsell nudge as a menu.
+	 *
+	 * @param object $nudge The $nudge object containing the content, CTA, link and tracks.
+	 */
+	public function add_upsell_nudge( $nudge ) {
+		$message = '
+<div class="upsell_banner">
+	<div class="banner__info">
+		<div class="banner__title">%1$s</div>
+	</div>
+	<div class="banner__action">
+		<button type="button" class="button">%2$s</button>
+	</div>
+</div>';
+
+		$message = sprintf(
+			$message,
+			wp_kses( $nudge['content'], array() ),
+			wp_kses( $nudge['cta'], array() )
+		);
+
+		add_menu_page( 'site-notices', $message, 'read', 'https://wordpress.com' . $nudge['link'], null, null, 1 );
+		add_filter( 'add_menu_classes', array( $this, 'set_site_notices_menu_class' ) );
+	}
+
+	/**
+	 * Adds a custom element class and id for Site Notices's menu item.
+	 *
+	 * @param array $menu Associative array of administration menu items.
+	 * @return array
+	 */
+	public function set_site_notices_menu_class( array $menu ) {
+		foreach ( $menu as $key => $menu_item ) {
+			if ( 'site-notices' !== $menu_item[3] ) {
+				continue;
+			}
+
+			$classes = ' toplevel_page_site-notices';
+
+			$menu[ $key ][4] = $menu_item[4] . $classes;
```

Comment:
Not that important: Can you add a check to see if the $menu_item[4] key exists? In <= PHP 7.4 this gives a notice, but from PHP 8 this will trigger a warning.

Reply:
Done e58d6f5